keps/2724-topology-aware-schedling/README.md

```diff
@@ -0,0 +1,569 @@
+about the node placement in a datacenter hierarchy using node labels.
+
+We also propose a set of APIs for Kueue administrators and users to utilize this
+information in order to optimize the network throughput between the pods.
+
+## Motivation
+
+It is common that AI / ML workloads require a significant amount of pod-to-pod
+communication to make progress. Thus, it is important, for runtime and overall
+cost, to make sure there is a high throughput network connection between the
+running pods.
+
+For example, some workloads may run twice slower if there is a pod placed on
+a node which belongs to a different block. However, currently the end user of
+Kueue has no way to "require" that all pods of its workload run within the same
+block.
+
+### Goals
+
+- allow a user to express that a workload can schedule only if all of its pods
+  can be placed on nodes which are close together (within the same level of
+  hierarchy)
+- allow a user to express that a workload prefers to schedule all of its pods
+  within the same level of hierarchy, but automatically relax the constraint if
+  not possible.
+
+### Non-Goals
+
+- support MultiKueue at the management cluster
+- support Cluster Autoscaler (ProvisioningRequest in particular)
+
+The above features might be pursued in the future in follow up KEPs.
+
+## Proposal
+
+* Introduce convention to represent the topological hierarchy as node labels
+* Provide API to configure the list of node labels representing levels in the
+  hierarchy per Resource Flavor
+* Introduce a set of Job annotations which allow to describe requirements for
+  TAS
+
+### User Stories
+
+#### Story 1
+
+As an ML researcher I run AI training workloads which require exchanging huge
+amounts of data between pods. The workloads take twice longer and are twice
+more expensive when there are pods running in different datacenter racks. I
+don't want to start my workload if the pods cannot be placed within the same
+rack.
+
+#### Story 2
+
+As an ML researcher I run AI training workloads which require communicating
+moderate amounts of data between pods. I would like to optimize the runtime of
+the workloads, by placing the pods as close to another as possible, ideally
+within a single rack, but I'm ok running the workload on nodes scattered across
+datacenter if placing all pods within the same hierarchy rack is not possible at
+the given time.
+
+### Notes/Constraints/Caveats (Optional)
+
+#### Integration support
+
+In the alpha iteration we aim to support Job & JobSet. If other integrations
+follow naturally we may as well support them in alpha.
+
+##### JobSet
+
+One complication we noticed for JobSet is that the proposed design assumes
+injecting the dedicated scheduling gate, called `kueue.x-k8s.io/topology` into
+the PodTemplate. However, currently JobSet's `spec.replicatedJob` field is
+immutable, even if the JobSet is suspended. This has been relaxed in JobSet 0.6
```

Comment:
if this is already merged, then you can change the wording here. Or is there anything else missing?

Reply:
Rephrased, thanks